Automattic/jetpack
# Settings  UI: allow non admins to use AtD, PbE and Publicize

Fixes #6475

#### Changes proposed in this Pull Request:

* allow non admin to use After the Deadline, Post by Email, and, if Publicize is active, display the Sharing tab. This tab is not visible when only Sharing is active because it gives non admins the ability to change the appearance of the site and that's only something admins should do.

* extend GUI tests to verify that a non admin can access After the Deadline settings group, the Post by Email settings card. If Publicize is active, display the Sharing tab.

* if  user is unlinked, hide Sharing tab and block PbE settings card

* don't show main toggles when user can't manage modules

* the `/settings` endpoint was updated. There was an issue where it wasn't allowing non-admins to set AtD options or regenerate PbE address.

* if user is subscriber, don't show the notice "You are currently running a development version of Jetpack"

* when AtD or PbE are disabled don't show their card in settings available to non admins.

* if AtD and PbE are disabled, a message explaining why the tab is empty will be shown to non admins.

* if non admin is unlinked, display a clickable card explaining that PbE is disabled because the account needs to be linked

#### Testing instructions:

* log in as a non-admin and make sure you can see AtD and PbE. If Publicize is active, you should see the Sharing tab.
* contributors must not have the Sharing tab available
* disable AtD or PbE (not both), the disabled feature should no longer display a card for non admins
* disable both AtD and PbE, cards should not be displayed for non admins and instead a message should be displayed
* with PbE active, if user is unlinked, a message should be displayed below the card prompting user to link.
